psi4/src/psi4/occ/cepa_iterations.cc

```diff
@@ -135,6 +139,108 @@ void OCCWave::cepa_iterations() {
+    outfile->Printf(" ============================================================================== \n");
+    outfile->Printf("\n");
+    outfile->Printf("  Iter    E_corr           E_total            DE           T2 RMS        \n");
+    outfile->Printf("  ----   -------------    ---------------    ----------   ----------    \n");
+
+    /********************************************************************************************/
+    /************************** NR iterations **************************************************/
+    /********************************************************************************************/
+    itr_occ = 0;
+    conver = 1;  // Assuming that the iterations will converge
+                 // DIIS
+    DIISManager t2_diis;
+    if (nooA + nooB != 1) {
+        if (reference_ == "RESTRICTED") {
+            dpdbuf4 T;
+            psio_->open(PSIF_OCC_DPD, PSIO_OPEN_OLD);
+            global_dpd_->buf4_init(&T, PSIF_OCC_DPD, 0, ID("[O,O]"), ID("[V,V]"), ID("[O,O]"), ID("[V,V]"), 0,
+                                   "T2 <OO|VV>");
+            t2_diis = DIISManager(maxdiis_, "CEPA DIIS T2 Amps", DIISManager::RemovalPolicy::LargestError, DIISManager::StoragePolicy::OnDisk);
+            t2_diis.set_error_vector_size(&T);
+            t2_diis.set_vector_size(&T);
+            global_dpd_->buf4_close(&T);
+            psio_->close(PSIF_OCC_DPD, 1);
+        }
+
+        else if (reference_ == "UNRESTRICTED") {
+            dpdbuf4 Taa, Tbb, Tab;
+            psio_->open(PSIF_OCC_DPD, PSIO_OPEN_OLD);
+            global_dpd_->buf4_init(&Taa, PSIF_OCC_DPD, 0, ID("[O,O]"), ID("[V,V]"), ID("[O,O]"), ID("[V,V]"), 0,
+                                   "T2 <OO|VV>");
+            global_dpd_->buf4_init(&Tbb, PSIF_OCC_DPD, 0, ID("[o,o]"), ID("[v,v]"), ID("[o,o]"), ID("[v,v]"), 0,
+                                   "T2 <oo|vv>");
+            global_dpd_->buf4_init(&Tab, PSIF_OCC_DPD, 0, ID("[O,o]"), ID("[V,v]"), ID("[O,o]"), ID("[V,v]"), 0,
+                                   "T2 <Oo|Vv>");
+            t2_diis = DIISManager(maxdiis_, "CEPA DIIS T2 Amps", DIISManager::RemovalPolicy::LargestError, DIISManager::StoragePolicy::InCore);
+            t2_diis.set_error_vector_size(&Taa, &Tbb, &Tab);
+            t2_diis.set_vector_size(&Taa, &Tbb, &Tab);
+            global_dpd_->buf4_close(&Taa);
+            global_dpd_->buf4_close(&Tbb);
+            global_dpd_->buf4_close(&Tab);
+            psio_->close(PSIF_OCC_DPD, 1);
+        }
+    }
+
+    // head of loop
+    do {
+        itr_occ++;
+        timer_on("T2");
+//        t2_amps();
+        t2_amps_remp(); // <- the only actual modification compared to regular CEPA(0)/D
+        timer_off("T2");
+        timer_on("CEPA Energy");
+        cepa_energy();
+        cepa_diis(t2_diis);    // <- CEPA diis can be reused without modifications
+        cepa_chemist();
```

Comment:
This is a good rationale for renaming functions, so that things labeled `cepa` are actually `cepa` only. I'd be thrilled if we could condense this and the CEPA code into a single function that just passes a flag around.

Reply:
for the future ...